neqo-transport/src/recovery.rs

```diff
@@ -126,18 +126,23 @@ impl LossRecoverySpace {
         self.space
     }
 
+    /// Find the time we sent the first packet that isn't yet declared lost.
     #[must_use]
     pub fn earliest_sent_time(&self) -> Option<Instant> {
-        // Lowest PN must have been sent earliest
-        let earliest = self.sent_packets.values().next().map(|sp| sp.time_sent);
-        debug_assert_eq!(
-            earliest,
-            self.sent_packets
-                .values()
-                .min_by_key(|sp| sp.time_sent)
-                .map(|sp| sp.time_sent)
-        );
-        earliest
+        // Sent packets should be ordered by packet number, so pick the first that isn't lost.
+        for sent in self.sent_packets.values() {
+            if !sent.lost() {
+                debug_assert_eq!(
```

Comment:
@grover, do you think we need to keep this assertion?  I fixed it, but I'm content in relying on the iteration order of the BTreeMap for this.

Reply:
I think I was worried we might switch to a different Map impl but that really would break a lot of other stuff, so this isn't really needed and can go.